### PR TITLE
Caching Module

### PR DESCRIPTION
**Pull request by tuxudo**
Created caching module and widget

### CODE REVIEW

File: app/modules/caching/caching_model.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /**
+     * Process data sent by postflight
+     *
+     * @param string data
+     *
+     **/
+    public function process($data)
+    {
+        // Delete previous entries
+        $this->deleteWhere('serial_number=?', $this->serial_number);
```
Comment (tuxudo) [inline app/modules/caching/caching_model.php]: The data is roughly 21 lines per caching hit or event. At its largest, it shouldn't be any larger than what a busy OS X Server uploads for the servermetrics module. The largest I've seen is roughly 512Kb.

**Comment by tuxudo** [comment]
This module needs a bit more work. It's not yet ready to be merged.

**Comment by tuxudo** [comment]
Updated Caching widget. It's as done as I can get it. I'm not able to get the graphing widget working, but it has a JSON API for outputting that data.

**Comment by bochoven** [comment]
Thanks, I don't have a caching server, so it is hard for me to do a proper review. I hope you will be available to fix things as needed :-)

**Pull request merged**